mithril-client-cli/src/commands/snapshot/download.rs

```diff
@@ -92,39 +92,35 @@ impl SnapshotDownloadCommand {
             .await?
             .with_context(|| format!("Can not get the snapshot for digest: '{}'", self.digest))?;
 
-        SnapshotDownloadCommand::check_local_disk_info(
-            1,
-            &progress_printer,
-            &db_dir,
-            &snapshot_message,
-        )?;
+        Self::check_local_disk_info(1, &progress_printer, &db_dir, &snapshot_message)?;
 
-        let certificate = SnapshotDownloadCommand::fetching_certificate_and_verifying_chain(
+        let certificate = Self::fetch_certificate_and_verifying_chain(
             2,
             &progress_printer,
             &client,
-            &snapshot_message,
+            &snapshot_message.certificate_hash,
         )
         .await?;
 
-        SnapshotDownloadCommand::downloading_and_unpacking_snapshot(
+        Self::download_and_unpack_snapshot(
             3,
             &progress_printer,
             &client,
             &snapshot_message,
             &db_dir,
         )
-        .await?;
+        .await
+        .with_context(|| {
+            format!(
+                "Can not get download and unpack snapshot for digest: '{}'",
+                self.digest
+            )
+        })?;
 
-        let message = SnapshotDownloadCommand::computing_snapshot_digest(
-            4,
-            &progress_printer,
-            &certificate,
-            &db_dir,
-        )
-        .await?;
+        let message =
+            Self::compute_snapshot_message(4, &progress_printer, &certificate, &db_dir).await?;
 
-        SnapshotDownloadCommand::verifying_snapshot_signature(
+        Self::verifying_snapshot_signature(
```

Comment:
You forgot this one :slightly_smiling_face: 
```suggestion
        Self::verify_snapshot_signature(
```

Reply:
Oops I'll fix that right away 🤓, thanks!